src/Storages/MySQL/MySQLSettings.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <Core/Defines.h>
+#include <Core/BaseSettings.h>
+
+
+namespace Poco::Util
+{
+    class AbstractConfiguration;
+}
+
+
+namespace DB
+{
+class ASTStorage;
+
+#define LIST_OF_MYSQL_SETTINGS(M) \
+    M(UInt64, connection_pool_size, 16, "Size of connection pool (if all connections are in use, the query will wait until some connection will be freed).", 0) \
+    M(UInt64, connection_max_tries, 3, "Number of retries for pool with failover", 0) \
+    \
+    M(Bool, connection_auto_close, true, "Auto-close connection after query execution, i.e. disable connection reuse.", 0) \
+    M(UInt64, num_tries_on_connection_loss, 5, "Number of tries on connection loss", 0) \
```

Comment:
@azat also not sure about this setting, I checked how it used right now, not sure that it is good on client level to retry connection, if we get connection from pool it should be connected and if there will be disconnect in the middle it is fine to throw exception, better to remove it from configuration for now.

Reply:
There should be the opposite use case - decrease it, since 5 (default) is quite big.

Reply:
@azat there should be no such setting at all, better to keep as is for now, and not expose it to client.